src/datasources/notifications/entities/notification-devices.entity.ts

```diff
@@ -0,0 +1,48 @@
+import { NotificationSubscription } from '@/datasources/notifications/entities/notification-subscription.entity';
+import type { UUID } from 'crypto';
+import {
+  Check,
+  Column,
+  Entity,
+  Unique,
+  OneToMany,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+
+@Entity('push_notification_devices')
+@Unique('device_uuid', ['device_uuid'])
+@Check('device_type', 'device_type IN ("ANDROID", "IOS", "WEB")')
+export class NotificationDevice {
```

Comment:
It can be done in a follow up, but I would be in favour of (on the domain level) validating these entities and, as such, we could then implement the type here, e.g.

```ts
// Datasource
export class NotificationDevice implements z.infer<NotificationDeviceSchema> { ... }

// Domain
function example() {
  const deviceFromDatabase = // ...
  return NotificationDeviceSchema.parse(device)
}
```

Reply:
I like the idea, but let's do it in the notification PR.